xstate-chatbot/nodejs/src/machine/vitals.js

```diff
@@ -1010,18 +1021,31 @@ const vitalsFlow = {
     addVitals: {
       id: 'addVitals',
       invoke: {
-        src: (context) => vitalsService.addVitals(context.user, context.slots.vitals),
-        onDone: {
+        src: (context) => vitalsService.addVitals(context.user, context.slots.vitals,context.slots.registerPatient),
+        onDone: [
+        {
+          cond: (context, event) =>context.slots.person.rrt==='NO',
           actions: assign((context, event) => {
             let mediaMessage = mediaUtil.createMediaMessage(`${config.staticMediaPath}/controlRoom_contact_numbers`, 'jpeg', undefined, '');
             dialog.sendMessage(context, dialog.get_message(messages.addVitals, context.user.locale), false);
             dialog.sendMessage(context, mediaMessage);
           }),
           target: '#endstate'
-        }
+        },
+        {
+          cond: (context, event) =>context.slots.person.rrt==='YES',
```

Comment:
When for both the conditions you want to do the same action, you don't need to write it separately.

Reply:
For both condition dialog.get_message is different